src/app/components/common/pagination/pagination.component.ts

```diff
@@ -1,66 +1,80 @@
 import { Component, EventEmitter, Input, OnChanges, OnInit, Output } from '@angular/core';
 
 @Component({
-    selector: 'app-pagination',
-    templateUrl: 'pagination.component.html'
+  selector: 'app-pagination',
+  templateUrl: 'pagination.component.html',
+  styleUrls: ['./pagination.component.css']
 })
 
 export class PaginationComponent implements OnInit, OnChanges {
-    @Input() page: number; // the current page
-    @Input() count: number; // how many total items there are in all pages
-    @Input() perPage: number; // how many items we want to show per page
-    @Input() totalPage: number;
+  @Input() page: number; // the current page
+  @Input() count: number; // how many total items there are in all pages
+  @Input() perPage: number; // how many items we want to show per page
+  @Input() totalPage: number;
 
-    @Output() goPrev = new EventEmitter<boolean>();
-    @Output() goNext = new EventEmitter<boolean>();
-    @Output() goPage = new EventEmitter<number>();
-    @Output() goFirst = new EventEmitter<boolean>();
-    @Output() goLast = new EventEmitter<number>();
+  @Output() goPrev = new EventEmitter<boolean>();
+  @Output() goNext = new EventEmitter<boolean>();
+  @Output() goPage = new EventEmitter<number>();
+  @Output() goFirst = new EventEmitter<boolean>();
+  @Output() goLast = new EventEmitter<number>();
 
-    middlePg: any;
+  @Output() onRangeChange = new EventEmitter<number>();
 
-    constructor() { }
+  middlePg: any;
 
-    ngOnInit() { }
+  paginationRange = ['10 Per Page', '20 Per Page', '30 Per Page', '50 Per Page'];
```

Comment:
change paginationRange  to paginationRange = ['10', '20', '30', '50']; and put 'Per Page' inside template like `{{range}}  Per Page`